src/core/net/ip6.cpp

```diff
@@ -127,6 +127,18 @@ uint8_t Ip6::PriorityToDscp(uint8_t aPriority)
     return dscp;
 }
 
+otError Ip6::GetPriority(const uint8_t *aDatagram, uint16_t aDatagramLen, uint8_t &aPriority) const
+{
+    otError error = OT_ERROR_NONE;
+    Header  header;
+
+    SuccessOrExit(error = header.Init(aDatagram, aDatagramLen));
```

Comment:
Is it possible to `reinterpret_cast` to `Header`?

Reply:
We need to check the validity of the IP6 header, so I prefer to use a function to initialize the header.

Reply:
Changed.